lib/rbs/inline/writer.rb

```diff
@@ -143,6 +144,41 @@ def translate_members(members, decl, rbs)
         end
       end
 
+      # @rbs members: Array[RBS::AST::Members::t | RBS::AST::Declarations::t]
+      # @rbs decl: AST::Declarations::ClassDecl | AST::Declarations::ModuleDecl
+      def unique_members(members, decl)
+        chosen = {} #: Hash[[Symbol, Class], RBS::AST::Members::InstanceVariable | RBS::AST::Members::ClassVariable | RBS::AST::Members::ClassInstanceVariable]
+        variables = [] #: Array[RBS::AST::Members::InstanceVariable | RBS::AST::Members::ClassVariable | RBS::AST::Members::ClassInstanceVariable]
+        members.each do |member|
+          case member
+          when RBS::AST::Members::InstanceVariable, RBS::AST::Members::ClassVariable, RBS::AST::Members::ClassInstanceVariable
+            variables << member
+          end
+        end
+
+        variables.each do |var|
+          if old = chosen[[var.name, var.class]]
+            if old.type == untyped && var.type != untyped
+              members.delete_if { _1.object_id == old.object_id }
+              chosen[[var.name, var.class]] = var
+            else
+              members.delete_if { _1.object_id == var.object_id }
```

Comment:
This code is a bit ugly. But we can't use `Array#delete` because it uses `#==` internally and removes other entries having the same name and same type.